query/src/lib/is-fetching.ts

```diff
@@ -0,0 +1,47 @@
+import { QueryFilters, notifyManager } from '@tanstack/query-core';
+import { injectQueryClient } from './query-client';
+import { DestroyRef, inject, Injectable, InjectionToken } from '@angular/core';
+import { distinctUntilChanged, Observable } from 'rxjs';
+import { takeUntilDestroyed, toSignal } from '@angular/core/rxjs-interop';
+
+@Injectable({ providedIn: 'root' })
+export class IsFetching {
+  private queryClient = injectQueryClient();
+  private destroyRef = inject(DestroyRef);
```

Comment:
destroyRef is redundant here because it's providedIn: 'root'. Let's remove it and let the consumer handle this.